backend/btrixcloud/pages.py

```diff
@@ -757,34 +733,40 @@ async def list_collection_pages(
             # default sort: seeds first, then by timestamp
             aggregate.extend([{"$sort": {"isSeed": -1, "ts": 1}}])
 
-        aggregate.extend(
-            [
-                {
-                    "$facet": {
-                        "items": [
-                            {"$skip": skip},
-                            {"$limit": page_size},
-                        ],
-                        "total": [{"$count": "count"}],
-                    }
-                },
-            ]
-        )
+        # aggregate.extend(
+        #    [
+        #        {
+        #            "$facet": {
+        #                "items": [
+        #                    {"$skip": skip},
+        #                    {"$limit": page_size},
+        #                ],
+        #                "total": [{"$count": "count"}],
+        #            }
+        #        },
+        #    ]
+        # )
+        if skip:
+            aggregate.append({"$skip": skip})
+        aggregate.append({"$limit": page_size})
 
         # Get total
         cursor = self.pages.aggregate(aggregate)
-        results = await cursor.to_list(length=1)
-        result = results[0]
-        items = result["items"]
+        results = await cursor.to_list(length=page_size)
+        items = results
+        # result = results[0]
+        # items = result["items"]
 
-        try:
-            total = int(result["total"][0]["count"])
-        except (IndexError, ValueError):
-            total = 0
+        # try:
+        # total = int(result["total"][0]["count"])
+        # except (IndexError, ValueError):
+        #    total = 0
+
+        total = await self.pages.estimated_document_count()
```

Comment:
Hm, I know this is faster, but won't it return the estimated total count of all pages, not an estimated total count of all the pages available with this specific filters, etc?

Reply:
removed, not including totals for this reason.